app/controllers/RiderRankingConnectionController.java

```diff
@@ -0,0 +1,116 @@
+package controllers;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import models.RiderRanking;
+import models.enums.RankingType;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import play.mvc.BodyParser;
+import play.mvc.Controller;
+import play.mvc.Result;
+import repository.interfaces.RiderRankingRepository;
+
+import javax.inject.Inject;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executors;
+
+import static play.libs.Json.toJson;
+
+public class RiderRankingConnectionController extends Controller {
+    private final RiderRankingRepository riderRankingRepository;
+
+    @Inject
+    public RiderRankingConnectionController(RiderRankingRepository riderRankingRepository) {
+        this.riderRankingRepository = riderRankingRepository;
+    }
+
+
+    public CompletionStage<Result> getRiderRankings(long riderStageConnectionId) {
+        return riderRankingRepository.getAllRiderRankings(riderStageConnectionId).thenApplyAsync(riderRankings -> {
+            return ok(toJson(riderRankings));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
+                case "IndexOutOfBoundsException":
+                    res = badRequest("No rider rankings are set in DB.");
+                    break;
+                default:
+                    res = internalServerError(ex.getMessage());
+            }
+            return res;
+        });
+    }
+
+    public CompletionStage<Result> getRiderRankingsByType(long riderStageConnectionId, String rankingType) {
+        return riderRankingRepository.getAllRiderRankingsByType(riderStageConnectionId, rankingType).thenApplyAsync(riderRankings -> {
+            return ok(toJson(riderRankings));
+        }).exceptionally(ex -> {
+            Result res = null;
+            switch (ExceptionUtils.getRootCause(ex).getClass().getSimpleName()){
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)